src/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/Base64Transforms.cs

```diff
@@ -0,0 +1,294 @@
+{
+    using System;
+    using Text;
+
+    public enum FromBase64TransformMode
+    {
+        IgnoreWhiteSpaces = 0,
+        DoNotIgnoreWhiteSpaces = 1,
+    }
+
+    public class ToBase64Transform : ICryptoTransform
+    {
+        // converting to Base64 takes 3 bytes input and generates 4 bytes output
+        public int InputBlockSize
+        {
+            get { return (3); }
+        }
+
+        public int OutputBlockSize
+        {
+            get { return (4); }
+        }
+
+        public bool CanTransformMultipleBlocks
+        {
+            get { return (false); }
+        }
+
+        public virtual bool CanReuseTransform
+        {
+            get { return (true); }
+        }
+
+        public int TransformBlock(byte[] inputBuffer, int inputOffset, int inputCount, byte[] outputBuffer, int outputOffset)
+        {
+            if (inputBuffer == null) throw new ArgumentNullException(nameof(inputBuffer));
+            if (inputOffset < 0) throw new ArgumentOutOfRangeException(nameof(inputOffset), SR.ArgumentOutOfRange_NeedNonNegNum);
+            if (inputCount < 0 || (inputCount > inputBuffer.Length)) throw new ArgumentException(SR.Argument_InvalidValue);
+            if ((inputBuffer.Length - inputCount) < inputOffset) throw new ArgumentException(SR.Argument_InvalidOffLen);
+
+            // For now, only convert 3 bytes to 4
+            char[] temp = new char[4];
+            Convert.ToBase64CharArray(inputBuffer, inputOffset, 3, temp, 0);
+
+            byte[] tempBytes = Encoding.ASCII.GetBytes(temp);
+            if (tempBytes.Length != 4) throw new CryptographicException(SR.Cryptography_SSE_InvalidDataSize);
+
+            Buffer.BlockCopy(tempBytes, 0, outputBuffer, outputOffset, tempBytes.Length);
+            return (tempBytes.Length);
+        }
+
+        public byte[] TransformFinalBlock(byte[] inputBuffer, int inputOffset, int inputCount)
+        {
+            // Do some validation
+            if (inputBuffer == null) throw new ArgumentNullException(nameof(inputBuffer));
+            if (inputOffset < 0) throw new ArgumentOutOfRangeException(nameof(inputOffset), SR.ArgumentOutOfRange_NeedNonNegNum);
+            if (inputCount < 0 || (inputCount > inputBuffer.Length)) throw new ArgumentException(SR.Argument_InvalidValue);
+            if ((inputBuffer.Length - inputCount) < inputOffset) throw new ArgumentException(SR.Argument_InvalidOffLen);
+
+            // Convert.ToBase64CharArray already does padding, so all we have to check is that
+            // the inputCount wasn't 0
+
+            // Again, for now only a block at a time
+            if (inputCount == 0)
+            {
+                return (Array.Empty<byte>());
+            }
+            else
+            {
+                char[] temp = new char[4];
+                Convert.ToBase64CharArray(inputBuffer, inputOffset, inputCount, temp, 0);
+                byte[] tempBytes = Encoding.ASCII.GetBytes(temp);
```

Comment:
(Weird, apparently when the UX said it lost my comment it really hadn't, and it showed up... sorry for the double-post)